torchtitan/distributed/pipeline.py

```diff
@@ -25,50 +26,73 @@
 # TODO: It's unclear if this API is general enough to be used by other models.
 # If not, we should move it to a Transformer-specific directory.
 def generate_split_points(
-    pipeline_parallel_schedule: str, pp_dim: int, num_layers: int
+    schedule_str: str,
+    layers_per_stage: Optional[int],
+    pp_dim: int,
+    num_layers: int,
+    input_output_weight: int = 1,
```

Comment:
In my internal use case, we use pipeline parallel with a heavy input block (i.e vision encoder) and normal Linear output block.
Currently we sweeps the PP config with different manual split points to balance the Memory.
So this setting while going in the right direction, wouldn't be generally enough to use.
Would you consider splitting input/output weight into separate arguments?
It would be easier to sweep by adjusting e.g `input_weight` etc.
```suggestion
    input_weight: int = 1,
    output_weight: int = 1,
```

Reply:
Hi @lkhphuc, thanks for the suggestion! This sounds makes sense to me, I will add it.

If you have a tool that is generalizable and does analysis to balance the memory, it would be cool to upstream it to the main PyTorch pipelining package (https://github.com/pytorch/pytorch/tree/main/torch/distributed/pipelining).